Source/WebCore/platform/mediastream/RealtimeMediaSource.cpp

```diff
@@ -1460,6 +1460,11 @@ void RealtimeMediaSource::getPhotoCapabilities(PhotoCapabilitiesHandler&& comple
     completion(PhotoCapabilitiesOrError("Not supported"_s));
 }
 
+void RealtimeMediaSource::getPhotoSettings(PhotoSettingsHandler&& completion)
```

Comment:
This would be a great application for using NativePromise ; much clearer error/rejection path.

Reply:
OK